perun-core/src/main/java/cz/metacentrum/perun/core/bl/AttributesManagerBl.java

```diff
@@ -4026,5 +4026,29 @@ public void mergeAttributesValues(PerunSession sess, Member member, List<Attribu
 	 */
 	void convertAttributeToUnique(PerunSession session, int attrId) throws InternalErrorException, AttributeNotExistsException, AttributeAlreadyMarkedUniqueException;
 
+	/**
+	 * Check if attribute is from the same namespace as it's handler
+	 *
+	 * @param sess
+	 * @param attributeDefinition
+	 * @param handler attribute's handler
+	 * @throws WrongAttributeAssignmentException
+	 * @throws InternalErrorException
+	 */
+	void checkAttributeAssignment(PerunSession sess, AttributeDefinition attributeDefinition, PerunBean handler) throws WrongAttributeAssignmentException, InternalErrorException;
+
+	/**
+	 * Check if attribute is from the same namespace as combination of perunBean handlers.
+	 * Order of handlers does not matter.
+	 *
+	 * @param sess perun session
+	 * @param attributeDefinition the attribute definition
+	 * @param handler1 one of attribute's handlers
+	 * @param handler2 one of attribute's handlers
+	 *
+	 * @throws WrongAttributeAssignmentException if assignment of attribute is correct
```

Comment:
You probably wanted to say when the assignment is **not** correct.

Reply:
Fixed.